src/config.js

```diff
@@ -118,3 +130,43 @@ export function loadJSConfigFile(filePath: string): Object {
   }
   return configObject;
 }
+
+type DiscoverConfigFilesParams = {|
+  getHomeDir: () => string,
+|};
+
+export async function discoverConfigFiles(
+  {getHomeDir = os.homedir}: DiscoverConfigFilesParams = {}
+): Promise<Array<string>> {
+  const magicConfigName = 'web-ext-config.js';
+
+  // Config files will be loaded in this order.
+  const possibleConfigs = [
+    // Look for a magic hidden config (preceded by dot) in home dir.
+    path.join(getHomeDir(), `.${magicConfigName}`),
+    // Look for a magic config in the current working directory.
+    path.join(process.cwd(), magicConfigName),
+  ];
+
+  const configs = await Promise.all(possibleConfigs.map(
+    async (fileName) => {
+      const resolvedFileName = path.resolve(fileName);
+      if (await fileExists(resolvedFileName)) {
+        return resolvedFileName;
+      } else {
+        log.debug(
+          `Discovered config "${resolvedFileName}" does not ` +
+          'exist or is not readable');
+        return false;
+      }
+    }
+  ));
+
+  const existingConfigs = [];
+  configs.forEach((f) => {
```

Comment:
I'll take another look but I think `filter()` was failing because Flow doesn't seem to understand that the returned array will not contain undefined values (or something like that).

Reply:
This is the best workaround I found:

```js
  const files = (
    (configs.filter((f) => typeof f === 'string'): Array<any>): Array<string>
  );
  return files;
```

That's too confusing though. I think it's probably best to leave it as is.

The Flow bug: https://github.com/facebook/flow/issues/1414

Reply:
yeah, I agree,  better to leave it as it is (which is also probably much more readable to a contributor that is not so used to the flow syntaxes).